hiedb.cabal

```diff
@@ -57,3 +57,28 @@ library
   hs-source-dirs:      src
   default-language:    Haskell2010
   ghc-options:         -Wall
+
+test-suite hiedb-tests
+    type:               exitcode-stdio-1.0
+    hs-source-dirs:     test
+    main-is:            Main.hs
+    build-tool-depends: hiedb:hiedb
+    other-modules:      Paths_hiedb
+    autogen-modules:    Paths_hiedb
+    build-depends:      base >= 4.12 && < 4.15
+                      , directory
+                      , hspec
+                      , process
+                      , temporary
+                      , test-lib
+    default-language:   Haskell2010
+
+-- Private library used in tests
+library test-lib
```

Comment:
[internal library](https://cabal.readthedocs.io/en/3.4/cabal-package.html#sublibs) - only tests will see this.
Tests depend on this, so it will be compiled before tests.
Tests will use the .hie dir produced by ghc-options below as the base directory for indexing

Reply:
switched away from internal library to calling ghc directly. I assume it's ok for the test to use the "ghc" currently present on PATH. or would you suggest some other approach to locating ghc?